src/libcore/ptr.rs

```diff
@@ -145,6 +148,31 @@ pub use intrinsics::write_bytes;
 /// // Ensure that the last item was dropped.
 /// assert!(weak.upgrade().is_none());
 /// ```
+///
+/// Drops a potentially unaligned value by copying it to aligned memory first:
+/// ```
+/// use std::ptr;
+/// use std::mem;
+///
+/// unsafe fn drop_after_copy<T>(to_drop: *mut T) {
+///     let mut copy: T = mem::uninitialized();
+///     let copy = &mut copy as *mut T;
+///     ptr::copy(to_drop, copy, 1);
+///     ptr::drop_in_place(copy);
+/// }
```

Comment:
I think one doesn't need `drop_in_place` here since (the first) `copy` will be dropped on scope exit.

Reply:
True... but that's because of `mem::uninitialized`. I'd prefer to use `MaybeUninit`.

Though then I guess this just changes to `drop(copy.into_inner())` so sure.

Reply:
I'd prefer for this to use `MaybeUninit` (or an `union`) as well.

Reply:
That would be blocking this on https://github.com/rust-lang/rust/pull/53508.